imagepuller/internal/service/verify.go

```diff
@@ -116,12 +124,18 @@ func (s *ImagePullerService) storeAndVerifyLayers(log *slog.Logger, remoteImg gc
 			return "", fmt.Errorf("closing layer reader: %w", err)
 		}
 
-		ldManifest := manifest.Layers[idx].Digest.String()
 		ld := putLayer.CompressedDigest.String()
 		if ldManifest != ld {
 			return "", fmt.Errorf("%w: expected digest '%s' but got digest '%s'", errValidateLayer, ldManifest, ld)
 		}
 
+		if err := s.Store.RemoveNames(putLayer.ID, putLayer.Names); err != nil {
+			return "", fmt.Errorf("removing pre-existing layer names: %w", err)
+		}
+		if err := s.Store.AddNames(putLayer.ID, []string{ldManifest}); err != nil {
+			return "", fmt.Errorf("adding layer digest as layer name: %w", err)
+		}
```

Comment:
What names is the library using, if not the digest? Could we simply set the name in the `Store.PutLayer` operation?

Reply:
My thinking here was originally that not stripping names might allow a bad actor to write the digest of layer B of image B as a name in layer A from image A, causing the imagepuller to create a hybrid image with some layers from B, some from A if B is pulled before A. If B had e.g. suid binaries in places accessible to an application running in A, that might open a way for privilege escalation.

Buuut your comment prompted me to look into it again, and since my assumption was faulty and the names are not a feature of docker images, but of the store, this all becomes irrelevant 😂 

We're doing a similar thing for wholy-cached images, want me to remove both instances?